src/big_core_input.cpp

```diff
@@ -40,54 +55,39 @@ void BigCoreInput::readFile(const std::string& filename, uint64_t bytes)
     // 2. volat metodu openFile(filename)
     // 3. volat metodu loadToMemory()
 
-	//this->setMemorySize(bytes);
-
-	//std::ifstream file(filename);
-	//if (!file)
-	//{
-	//	throw "Unable to open file for reading!";
-	//}
-
-	//char buffer[CHUNK_LENGTH];
-
-	//file.read(buffer, CHUNK_LENGTH);
-	//if (!this->checkHeader(buffer))
-	//{
-	//	throw "File format is not BIG format!";
-	//}
-
-	//uint64_t id, length;
-	//while (!file.eof())
-	//{
-	//	if (!this->readChunk(file, id, length) && !this->readData(file, id, length))
-	//	{
-	//		throw "Error while reading file!";
-	//	}
-	//}
-	//this->lock = true;
+	this->setMemorySize(bytes);
+	this->openFile(filename);
+	this->loadToMemory(filename);
 }
 
-void BigCoreInput::loadToMemory()
+void BigCoreInput::loadToMemory(const std::string& filename)
 {
     // RV:
     // 1. overit, ze se data vejdou do pameti, else return
     // 2. alokovat prislusne velkou pamet
     // 3. posunout se v souboru na misto, kde zacinaji data (metoda openFile ho ulozila do promenne)
     // 4. nacist data
 
-    //if (this->dataLength + length <= this->memorySize)
-    //{
-    //    // allocate new memory and copy old data to it
-    //    char* data = new char[this->dataLength + length];
-    //    std::copy(this->data, this->data + this->dataLength, data);
-
-    //    // clear old data and assign new ones
-    //    this->clear();
-    //    this->data = data;
+	std::ifstream file(filename, std::ios_base::binary);
+	uint64_t length;
+	file.seekg(this->dataPosition - CHUNK_LENGTH);
+	file.read((char*)&length, CHUNK_LENGTH);
 
-    //    // move pointer past already loaded data and load new ones
-    //    file.read(this->data + this->dataLength, length);
-    //}
+    if (this->dataLength + length <= this->memorySize)
```

Comment:
Tady trochu nechapu. Melo by se nejdrive volat this->clear(); a pote jen overit, ze length <= memorySize. Pridavat k soucasnym datum nejaka dalsi data mi nedava smysl. Kdyz napriklad pouziju readFile na jeden soubor a potom readFile na druhy soubor, mel bych mit pouze parametry a data druheho souboru, vse co bylo v prvnim se zahodi pred ctenim druheho. Jinak by to byl pekny gulas a nedali bychom to nikdy dohromady. Respektive this->clear(); by se melo volat uz v openFile kdyz oteviram novy soubor, protoze data ze stareho uz mi jsou v tu chvili k nicemu.